src/opm/input/eclipse/Schedule/MSW/WellSegments.cpp

```diff
@@ -76,6 +84,26 @@ namespace Opm {
         return this->m_segments.empty();
     }
 
+    int WellSegments::maxSegmentID() const
+    {
+        return std::accumulate(this->m_segments.begin(),
+                               this->m_segments.end(), 0,
+            [](const int maxID, const Segment& seg)
+        {
+            return std::max(maxID, seg.segmentNumber());
+        });
```

Comment:
Of course this is correct, but it seems strange to use accumulate when not accumulating anything. Any particular reason not to use `max_element` with a custom Comparator
```
if (this.>m_segments.empty())
    return 0;
return std::max_element(this->m_segments.begin(),
                        this->m_segments.end(),
                        [](const Segment& seg1, const Segment& seg1 ){
                            return seg1.segmentNumber() < seg2..segmentNumber();
                        })->segmentNumber();
```
(should be at least as performant) or at least `for_each` with capturing maxID by reference?
```
int maxID = 0;
std::for_each(this->m_segments.begin(),
             this->m_segments.end(),
             [&maxID](const Segment& seg)
             {
                 maxID = std::max(maxID, seg.segmentNumber());
             });
```

Reply:
BTW: I am merging this anyway...